src/Controls/src/Core/Page/Page.cs

```diff
@@ -853,5 +875,13 @@ internal void SendNavigatedFrom(NavigatedFromEventArgs args)
 		/// <returns>The <see cref="Window"/> instance that parents the page.</returns>
 		public virtual Window GetParentWindow()
 			=> this.FindParentOfType<Window>();
+
+		/// <summary>
+		/// Adds inset to the page if the page is a part of a shell
+		/// </summary>
+		void IShellContentInsetObserver.OnInsetChanged(Thickness inset, double tabThickness)
+		{
+			Padding = new Thickness(inset.Left, inset.Top + tabThickness, inset.Right, inset.Bottom);
```

Comment:
Yea, I don't know if we can do this. 

We intentionally got rid of the padding path. 
With NET8 we set the additional insets 
https://github.com/dotnet/maui/blob/0da0a61a7e3f127bcab8a800dfeda4101a0d15bc/src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/ShellSectionRootRenderer.cs#L553-L613

So someone could read from the SafeAreaInsets property to get the value and then use that for padding if they want to.